File: services/payment/src/main/java/com/example/iter/payment/dto/request/PaymentHistorySearchRequest.java
package com.example.iter.payment.dto.request;

import com.example.iter.payment.api.PaymentStatus;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;

public record PaymentHistorySearchRequest(
        PaymentStatus status,

        @Min(value = 0, message = "페이지 번호는 0 이상이어야 합니다.")
        Integer page,

        @Min(value = 1, message = "페이지 크기는 1 이상이어야 합니다.")
        @Max(value = 100, message = "페이지 크기는 100 이하여야 합니다.")
        Integer size
) {
    public PaymentHistorySearchRequest {
        page = page == null ? 0 : page;
        size = size == null ? 20 : size;
    }
}

File: services/payment/src/main/kotlin/iter/payment/dto/request/PaymentHistorySearchRequest.kt
package iter.payment.dto.request

import iter.payment.api.PaymentStatus
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

class PaymentHistorySearchRequest(
    val status: PaymentStatus?,
    page: Int?,
    size: Int?,
) {
    @field:Min(value = 0, message = "페이지 번호는 0 이상이어야 합니다.")
    val page: Int = page ?: 0

    @field:Min(value = 1, message = "페이지 크기는 1 이상이어야 합니다.")
    @field:Max(value = 100, message = "페이지 크기는 100 이하여야 합니다.")
    val size: Int = size ?: 20
}
